src/Microsoft.AspNetCore.Cors/Infrastructure/CorsPolicyBuilder.cs

```diff
@@ -91,6 +91,18 @@ public CorsPolicyBuilder WithMethods(params string[] methods)
         }
 
         /// <summary>
+        /// Adds the specified <paramref name="isOriginAllowed"/> to the policy.
+        /// </summary>
+        /// <param name="isOriginAllowed">The function used by the policy to evaluate if an origin is allowed.</param>
+        /// <returns>The current policy builder</returns>
+        public CorsPolicyBuilder WithIsOriginAllowed(Func<string, bool> isOriginAllowed)
+        {
+            _policy.IsOriginAllowed = isOriginAllowed;
```

Comment:
The `WithXyz` methods all *combine* logic, whereas this method is a *destructive* method that replaces a setting. Would a better name for this method be something like `SetIsOriginAllowed(...)`?